### ISSUE TITLE
[Feature] Keep QR code/address in-view when selecting/scrolling between subaddresses

### ISSUE THREAD

**Issue by sethsimmons** [cake-tech/cake_wallet#173]
When you have more than 5 or so subaddresses, selecting them and seeing the associated address becomes quite cumbersome as you have to scroll down, select the subaddress, scroll up, check the address, etc.

Would be great for the list of addresses to scroll as *just* the bottom portion of the screen, keeping the QR code and address itself in view even when scrolling/selecting different addresses.

Could even have QR/address minimize to a smaller portion of the screen, as long as the address itself stay visible while scrolling/selecting different addresses.

**Comment by SamsungGalaxyPlayer** [comment]
Can you explain your desired flow a bit more? We don't normally expect users to pay attention to the specific address strings; that's what the address labels are for. The current flow allows for users to have a full display of available address labels, and select the proper one. Then they can scroll back up to see the address.

Would a button to jump back to the top assist with still showing many address labels?

The reason to keep the address and QR code visible while scrolling through and selecting different address labels would be if you were trying to stay on the same screen and accept payments from several different people or something.

**Comment by sethforprivacy** [comment]
The issue here is really three-fold:

1. Addresses in the list show *neither* a name or an address by default, so until you name each address you're just scrolling through blank boxes.
2. Attempting to name addresses according to the address (for instance after restoring a wallet, wanting to name the same addresses the same thing again) is a nightmare, as you have to scroll through a list of blank boxes, click on one randomly, scroll back up, check the address, then repeat until you find the right address.
3. You can't rename the address you've selected, so you have to click off of it and then hope you select the correct blank box to rename

There are really two ways to fix this IMO:

1. Name addresses a shortened form of the underlying address by-default (i.e. 8sssfdsf...323kklll) so users can recognize addresses they want to rename
2. Keep the address (not the QR code) visible at the top when scrolling/selecting so people can jump around to find the right one to rename

**Comment by sethforprivacy** [comment]
The biggest issue is really that once you have more than a few subaddresses used, the management and naming of them is a very bad user experience.

And this poor user experience will be repeated anytime a user restores a wallet with existing subaddresses, as subaddress names aren't part of the seed restore (of course).

**Comment by SamsungGalaxyPlayer** [comment]
This specific grievance seems to depend on a wallet restore. Cake doesn't allow people to generate addresses without a label within the app, so unless they are purposefully naming wallets as spaces only, this won't happen.

The best way forward as I see it is to fill in part of the address as you described for restored wallets, instead of having them be blank.

**Comment by sethforprivacy** [comment]
> This specific grievance seems to depend on a wallet restore. Cake doesn't allow people to generate addresses without a label within the app, so unless they are purposefully naming wallets as spaces only, this won't happen.
> 
> The best way forward as I see it is to fill in part of the address as you described for restored wallets, instead of having them be blank.

Yes, that would be an excellent approach and would solve the UX issues around restored wallets. I gave up on naming important addresses in Cake because of this issue, so I'll be a very happy camper to be able to easily identify restored addresses by an address snippet :)

**Comment by SamsungGalaxyPlayer** [comment]
Okay, I'm going to make an issue for that then.

**Comment by sethforprivacy** [comment]
For reference, this is what it currently looks like when you restore a wallet with many used subaddresses:

![image](https://user-images.githubusercontent.com/40500387/152224305-9eaa015e-9042-4fad-a980-688fbe625546.png)
